tests/XFTPCLI.hs

```diff
@@ -99,4 +102,37 @@ testXFTPCLISendReceive2servers = withXFTPServer . withXFTPServer2 $ do
         _ -> print srvs >> error "more than 2 servers returned"
       xftp ["recv", fd, recipientFiles, "--tmp=tests/tmp"]
         `shouldReturn` ["File received: " <> recipientFiles </> fileName]
-      LB.readFile (recipientFiles </> fileName) `shouldReturn` file
\ No newline at end of file
+      LB.readFile (recipientFiles </> fileName) `shouldReturn` file
+
+testXFTPCLIDelete :: IO ()
+testXFTPCLIDelete = withXFTPServer . withXFTPServer2 $ do
+  let filePath = senderFiles </> "testfile"
+  xftp ["rand", filePath, "19mb"] `shouldReturn` ["File created: " <> filePath]
+  file <- LB.readFile filePath
+  getFileSize filePath `shouldReturn` 19 * mb
+  let fdRcv1 = filePath <> ".xftp" </> "rcv1.xftp"
+      fdRcv2 = filePath <> ".xftp" </> "rcv2.xftp"
+      fdSnd = filePath <> ".xftp" </> "snd.xftp.private"
+  xftp ["send", filePath, senderFiles, "-n", "2", "-s", testXFTPServerStr <> ";" <> testXFTPServerStr2, "--tmp=tests/tmp"]
+    `shouldReturn` [ "File uploaded!",
+                     "Pass file descriptions to the recipient(s):",
+                     fdRcv1,
+                     fdRcv2,
+                     "Sender file description:",
+                     fdSnd
+                   ]
+  xftp ["del", fdRcv1]
+    `shouldThrow` anyException
+  xftp ["recv", fdRcv1, recipientFiles, "--tmp=tests/tmp"]
+    `shouldReturn` ["File received: " <> recipientFiles </> "testfile"]
+  LB.readFile (recipientFiles </> "testfile") `shouldReturn` file
+  listDirectory xftpServerFiles >>= \fs -> fs `shouldNotBe` []
```

Comment:
You can check file count instead

Reply:
I didn't find a more appropriate function